src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchService.kt

```diff
@@ -112,30 +111,26 @@ class MatchService(
 
 }
 
-fun BoolQueryBuilder.mustWhenPresent(query: String, value: String?): BoolQueryBuilder {
-  value.takeIf { !it.isNullOrBlank() }?.let {
-    this.must(matchQuery(query, it))
-  }
-  return this
-}
-
-fun BoolQueryBuilder.shouldWhenPresent(query: String, value: String?): BoolQueryBuilder {
-  value.takeIf { !it.isNullOrBlank() }?.let {
-    this.should(matchQuery(query, it))
-  }
-  return this
-}
-
-fun BoolQueryBuilder.mustWhenPresent(query: String, value: LocalDate?): BoolQueryBuilder {
-  value?.let {
+fun BoolQueryBuilder.mustWhenPresent(query: String, value: Any?): BoolQueryBuilder {
+  value.takeIf {
+    when(it) {
+      is String -> it.isNotBlank()
```

Comment:
rather than have 2 version of this API just squashed into one and do an additional check if a String